### PR TITLE
fix fuzzers

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Recent updates means the utils fuzzer is no longer relevant and one of the target functions for the strings fuzzer has been removed.

### CODE REVIEW

File: fuzzers/strings_fuzzer.c [hunk 1]
```diff
@@ -28,7 +29,6 @@ LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
   memcpy(ns, data, size);
   ns[size] = '\0';
 
-  pcmk_numeric_strcasecmp(ns, ns);
```
Comment (kgaillot) [inline fuzzers/strings_fuzzer.c:31]: The function was renamed with the first underbar doubled (`pcmk__numeric_strcasecmp`). In Pacemaker a single first underbar means public API and a double means internal-only, and for the next upstream release we're pulling a lot of the existing API internal.

I"ll have to remember to start checking the fuzzers when we make these changes ...

File: fuzzers/utils_fuzzer.c [hunk 2]
```diff
@@ -1,32 +0,0 @@
...
-{
-  char *ns;
-  if (size < 10) {
-    return 0;
-  }
-  ns = malloc(size+1);
-  memcpy(ns, data, size);
-  ns[size] = '\0';
-   
-  crm_parse_interval_spec(ns);
```
Comment (kgaillot) [inline fuzzers/utils_fuzzer.c:28]: This has been renamed to `pcmk_parse_interval_spec()`. (Another thing we're doing is renaming old APIs with the current convention of using a `pcmk` prefix.)

**Review by kgaillot — COMMENTED** [review]
It won't be as bad as you thought :)

File: fuzzers/utils_fuzzer.c (hunk 2 above)
Comment (DavidKorczynski) [reply fuzzers/utils_fuzzer.c:28]: Ah awesome, fixed

File: fuzzers/strings_fuzzer.c (hunk 1 above)
Comment (DavidKorczynski) [reply fuzzers/strings_fuzzer.c:31]: Great, thanks! Fixed

**Comment by DavidKorczynski** [comment]
> I"ll have to remember to start checking the fuzzers when we make these changes ...

@kgaillot we could enable the OSS-Fuzz set up to do a quick build/run in the CI? in this way we can (1) ensure fuzzers don't break and (2) catch regressions before they're merged

File: fuzzers/utils_fuzzer.c [hunk 3]
```diff
@@ -24,8 +24,9 @@ LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
   ns = malloc(size+1);
   memcpy(ns, data, size);
   ns[size] = '\0';
-   
-  crm_parse_interval_spec(ns);
+
+  guint result;
```
Comment (kgaillot) [inline fuzzers/utils_fuzzer.c:28]: C99 (declarations have to be before code)

**Review by kgaillot — COMMENTED** [review]
> @kgaillot we could enable the OSS-Fuzz set up to do a quick build/run in the CI? in this way we can (1) ensure fuzzers don't break and (2) catch regressions before they're merged

Hmm, not sure. Our CI has around three dozen OS+arch combinations. We can special-case things to run on a subset, but we try to avoid such things to keep maintenance simple. Maybe if more projects get added to fuzzing it would be worth it.

@fabbione FYI

File: fuzzers/utils_fuzzer.c (hunk 3 above)
Comment (DavidKorczynski) [reply fuzzers/utils_fuzzer.c]: Thanks, done!

**Comment by DavidKorczynski** [comment]
> Hmm, not sure. Our CI has around three dozen OS+arch combinations. We can special-case things to run on a subset, but we try to avoid such things to keep maintenance simple. Maybe if more projects get added to fuzzing it would be worth it.

Got it -- in either case it's easy to set up since there already is an OSS-Fuzz integration. Documentation is here (if you'd like to add it you can ping me and I'll set it up) https://google.github.io/oss-fuzz/getting-started/continuous-integration/

**Comment by kgaillot** [comment]
> > Hmm, not sure. Our CI has around three dozen OS+arch combinations. We can special-case things to run on a subset, but we try to avoid such things to keep maintenance simple. Maybe if more projects get added to fuzzing it would be worth it.
> 
> Got it -- in either case it's easy to set up since there already is an OSS-Fuzz integration. Documentation is here (if you'd like to add it you can ping me and I'll set it up) https://google.github.io/oss-fuzz/getting-started/continuous-integration/

Ah, this runs on your servers? I thought we'd have to self-host it on our existing CI. Yes, that could be worthwhile for the main branch.

**Pull request merged**